central/complianceoperator/v2/profiles/service/service_impl.go

```diff
@@ -79,26 +75,37 @@ func (s *serviceImpl) GetComplianceProfile(ctx context.Context, req *v2.Resource
 }
 
 // ListComplianceProfiles returns profiles matching given query
-func (s *serviceImpl) ListComplianceProfiles(ctx context.Context, query *v2.RawQuery) (*v2.ListComplianceProfilesResponse, error) {
-	// Fill in Query.
-	parsedQuery, err := search.ParseQuery(query.GetQuery(), search.MatchAllIfEmpty())
-	if err != nil {
-		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to parse query %v", err)
+func (s *serviceImpl) ListComplianceProfiles(ctx context.Context, request *v2.ProfilesForClusterRequest) (*v2.ListComplianceProfilesResponse, error) {
+	if request.GetClusterId() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "cluster is required")
 	}
 
-	// Fill in pagination.
-	paginated.FillPaginationV2(parsedQuery, query.GetPagination(), maxPaginationLimit)
-
-	profiles, err := s.complianceProfilesDS.SearchProfiles(ctx, parsedQuery)
+	profiles, err := s.complianceProfilesDS.GetProfilesByCluster(ctx, request.GetClusterId())
 	if err != nil {
-		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance profiles for query %v", query)
+		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance profiles for cluster %v", request.GetClusterId())
 	}
 
 	return &v2.ListComplianceProfilesResponse{
 		Profiles: storagetov2.ComplianceV2Profiles(profiles),
 	}, nil
 }
 
+// ListProfileSummaries returns profile summaries matching incoming clusters
+func (s *serviceImpl) ListProfileSummaries(ctx context.Context, request *v2.ClustersProfileSummaryRequest) (*v2.ListComplianceProfileSummaryResponse, error) {
+	if len(request.GetClusterIds()) == 0 {
```

Comment:
Based on the documentation - it looks like the http implementation expects a list of strings:

```
 $ ./bin/roxcurl "/v2/compliance/summary/profiles?clusterIds=916d89af-78a5-4f2a-8c4c-ae146b091f7a" | jq .
{
  "profiles": []
}
```

If one of the elements of the array is invalid, we're not going to generate a partial response? All clusters must exist for central to produce a response?

```
 $ ./bin/roxcurl "/v2/compliance/summary/profiles?clusterIds=916d89af-78a5-4f2a-8c4c-ae146b091f7a,foo" | jq .
{
  "error": "Unable to retrieve compliance profiles for clusters [916d89af-78a5-4f2a-8c4c-ae146b091f7a,foo]: invalid arguments",
  "code": 3,
  "message": "Unable to retrieve compliance profiles for clusters [916d89af-78a5-4f2a-8c4c-ae146b091f7a,foo]: invalid arguments",
  "details": []
}
```

Reply:
The intent behind this one was for providing the list of profiles to choose from when creating a schedule.  So the intent was to only show profiles that applied to ALL selected clusters.  So I think it is OK to not return a partial response if the data in the request is bad.  At least for now.  If we feel we want to deal with that differently, we can write a ticket and address it separately.